tidy3d/components/simulation.py

```diff
@@ -180,6 +180,13 @@ class Simulation(Box):  # pylint:disable=too-many-public-methods
         "based on structure definition, resulting in much higher accuracy for a given grid size.",
     )
 
+    normalize_index: Optional[pydantic.NonNegativeInt] = pydantic.Field(
+        0,
+        title="Normalization index",
+        description="Index of the source in the list of sources whose spectrum will be used to "
```

Comment:
"tuple" of sources, maybe

Reply:
Also, add what happens if it's None?

Reply:
> * Some of the validation logic (and None handling) is a little hard to follow, so let's just make sure we're testing all the edge cases.  For example, it seemed the `symmetry` might have never been applied because of that tiny bug.

Right, I'm pretty sure the symmetry would have been caught by the backend tests (I'm yet to make this work with the backend), but I'll add more cases to the frontend tests too.

> * Sometimes `sim_data[name]` returns a copy (if symmetry has an effect) and sometimes it doesn't.  I wonder whether this inconsistency could create some confusion for the user.  Example: user doesn't understand why their modifications of `sim_data[mode].amps` modify the actual data whereas modifying `sim_data[field].Ex` has no lasting effect. At the very least, we should be very clear when things are copied, maybe even throw a warning in that case?

Yeah I was wondering about that too, but it seems redundant to make a copy without symmetry (and with symmetry, the original data is at least twice smaller, so it's not so bad). For your example, I guess we generally don't intend people to manually change the data, but I do agree it's not great to have the inconsistency. So you're suggesting raising a warning in the symmetry case when a copy is made? Or how else can we be very clear ( just in the docstrings?)

Reply:
I'm not really sure, I think warning and docstrings is a good start, but may not be enough? is it problematic to apply symmetry when we load from file? and then the `SimulationData` in memory has symmetry (and normalization) both applied already and we won't have to worry about copying at all. Just a thought. 

Otherwise, our only other option (if we want to make things consistent) seems to be to copy everything, including when symmetry isn't applied.

1. Copy only when symmetry applied, warn and put note in docstrings.
2. Apply symmetry to everything when loading from file, copy nothing when accessing data.
3. Copy everything (like before).